### PR TITLE
error messages task #19

### PR DESCRIPTION
**Pull request by blackout314**
task #19 

add error messages on fail

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#53](https://github.com/zold-io/zold.github.io/issues/53) is now in scope, role is `REV`

File: js/map.js [hunk 1]
```diff
@@ -107,7 +112,10 @@ function put_marker_by_host(map, coords, host, port) {
     ip = $.grep(json.answer, function (a, i) { return a.type == 'A'; })[0].rdata;
     console.log('Host ' + host + ' resolved to ' + ip);
     put_marker_by_ip(map, coords, ip, port);
-  }).fail(function() { console.log('Failed to find IP for ' + host); });
+  }).fail(function() {
+    console.log('Failed to find IP for ' + host);
+    $messageContainer.append('<div class=""message-warning">Failed to find IP for ' + host + '</div>');
```
Comment (yegor256) [inline js/map.js]: @blackout314 this approach will lead to a serious overflow of the error message container

**Comment by yegor256** [comment]
@blackout314 looks like a rather dirty approach. In a few minutes/hours of work the message container will be full of messages. Will look ugly and will work slow.

**Comment by blackout314** [comment]
@yegor256 i can write a timer for delete it, or a cross for remove

**Comment by yegor256** [comment]
@blackout314 maybe, I'm not sure. But just dropping `div`s into a container is a very dirty approach, I think.

**Comment by blackout314** [comment]
@yegor256 like nodes dropped in table

**Comment by yegor256** [comment]
@blackout314 yes, but the amount of nodes is limited. Messages, on the other hand, may exist in thousands, if the web page stays open for an hour, for example.

**Comment by yegor256** [comment]
@blackout314 maybe we can drop them there and then check the length of the list of errors. If there are more than, say, 8, we should remove the oldest ones.

**Comment by blackout314** [comment]
@yegor256 ok

**Comment by blackout314** [comment]
@yegor256 i hope do better in this task
i refactored messages system and cleaning code

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#53](https://github.com/zold-io/zold.github.io/issues/53) is already in scope

File: js/map.js [hunk 2]
```diff
@@ -112,27 +117,47 @@ function put_markers(map, remotes) {
...
     console.log('Marker set for ' + coords + ' at ' + lat + '/' + lon);
-  }).fail(function() { console.log('Failed to find geo-location for ' + ip); });
+  }).fail(function() {
+    createMessage('Failed to find geo-location for ' + ip);
+  });
+}
+
+function createMessage(message) {
+  messagesRepository.push(message);
+	printMessages(messagesRepository.reverse());
```
Comment (yegor256) [inline js/map.js]: @blackout314 what's up with the indentation?

File: js/map.js [hunk 3]
```diff
@@ -112,27 +117,47 @@ function put_markers(map, remotes) {
...
+
+function printMessages(messages) {
+	var $messageContainer = $('.messages-container');
+  $messageContainer.html('');
+	messages.slice(0,8)
+    .forEach(function(message) {
+      console.log(message);
+    	$messageContainer.append('<div class="message-warning">' + message + '</div>');
+    });
+}
```
Comment (yegor256) [inline js/map.js]: @blackout314 what's going on with indentation here?

File: js/map.js [hunk 4]
```diff
@@ -112,27 +117,47 @@ function put_markers(map, remotes) {
...
     });
     console.log('Marker set for ' + coords + ' at ' + lat + '/' + lon);
-  }).fail(function() { console.log('Failed to find geo-location for ' + ip); });
+  }).fail(function() {
+    createMessage('Failed to find geo-location for ' + ip);
+  });
+}
+
+function createMessage(message) {
+  messagesRepository.push(message);
```
Comment (yegor256) [inline js/map.js]: @blackout314 don't you worry about memory overflow? This array will grow and grow.

**Comment by blackout314** [comment]
@yegor256 a fiddle for testing repository http://jsfiddle.net/1no7kqty/7/

**Comment by yegor256** [comment]
@blackout314 dude, indentation here is crazy :) What's wrong?

File: js/map.js (hunk 3 above)
Comment (blackout314) [reply js/map.js]: @yegor256 https://i.imgur.com/g3JmUQZ.png correct way for indent

**Comment by blackout314** [comment]
@yegor256 ok, array sliced every time to size=8
change indent

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:zold-io/zold.github.io#53` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#53](https://github.com/zold-io/zold.github.io/issues/53) is now out of scope

**Comment by 0crat** [comment]
The job is not in WBS, won't close the order

**Comment by 0crat** [comment]
Job `gh:zold-io/zold.github.io#53` is not assigned, can't get performer

**Pull request merged**